src/actions/isNetworkDomainAvailable.ts

```diff
@@ -0,0 +1,24 @@
+import { domainNameToId, getLogger } from "../utilities";
+import { ZNSHub } from "../contracts/types";
+import { checkContentModeration } from "../api/actions";
+
+const logger = getLogger("actions:getDomainMetadata");
+
+export const isNetworkDomainAvailable = async (
+  name: string,
+  hub: ZNSHub,
+  apiUri: string
+): Promise<boolean> => {
+    logger.trace(`Checking network domain availability for: ${name}`);
```

Comment:
Make sure to check for invalid characters here as well.

Reply:
Its baked into the checkContentModeration function